apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/common/ClaimDao.java

```diff
@@ -68,6 +84,75 @@ public Object getEntityById(ResourceTypeV2<?> type, String id) {
     return claimEntity;
   }
 
+  public <T> List<T> findAllByMbiHash(
+      Class<T> entityClass, String mbiHash, DateRangeParam lastUpdated) {
+    return findAllByAttribute(entityClass, "mbiHash", mbiHash, lastUpdated);
+  }
+
+  public <T> List<T> findAllByMbi(Class<T> entityClass, String mbi, DateRangeParam lastUpdated) {
+    return findAllByAttribute(entityClass, "mbi", mbi, lastUpdated);
+  }
+
+  @VisibleForTesting
+  <T> List<T> findAllByAttribute(
+      Class<T> entityClass,
+      String attributeName,
+      String attributeValue,
+      DateRangeParam lastUpdated) {
+    List<T> claimEntities = null;
+
+    CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<T> criteria = builder.createQuery(entityClass);
+    Root<T> root = criteria.from(entityClass);
+
+    criteria.select(root);
+    criteria.where(
+        builder.and(
+            builder.equal(root.get(attributeName), attributeValue),
+            lastUpdated == null
+                ? builder.and()
+                : createDateRangePredicate(root.get("lastUpdated"), lastUpdated, builder)));
+
+    Timer.Context timerClaimQuery = metricRegistry.timer(CLAIM_BY_MBI_METRIC_NAME).time();
+    try {
+      claimEntities = entityManager.createQuery(criteria).getResultList();
+    } finally {
+      long claimByIdQueryNanoSeconds = timerClaimQuery.stop();
+      TransformerUtilsV2.recordQueryInMdc(
+          CLAIM_BY_MBI_METRIC_QUERY,
+          claimByIdQueryNanoSeconds,
+          claimEntities == null || claimEntities.isEmpty() ? 0 : 1);
+    }
+
+    return claimEntities;
+  }
+
+  @VisibleForTesting
+  Predicate createDateRangePredicate(
+      Path<Instant> datePath, DateRangeParam dateRange, CriteriaBuilder builder) {
+    Instant from =
```

Comment:
If one of the bounds is empty, can we just skip building a predicate for it?  I think adding a where clause to these queries that makes a comparison where every row will return True will just make the query slower than it needs to be.

Reply:
The code that invokes `createDateRangePredicate()` checks if the any value is in `lastUpdated`, so if no bounds are set it does nothing.  However, do we want to support just one bound? I.e. if upper bound is set, we're just returning everything before date X, or if only lower bound is set, we're returning everything after date X?

Reply:
yeah thats what I was thinking, I don't think we need an effective start and effective end date as using those in the query will make it unnecessarily slower